tests/unit/test_elements.py

```diff
@@ -271,7 +271,7 @@ class TestCacheStep:
         ],
         ids=["str", "Path"],
     )
-    def test_ctor(self, paths: list[Union[str, Path]]):
```

Comment:
Alternatively we could consider dropping support for Python 3.8?

Reply:
Yes, let's do that in a separate PR.